### PR TITLE
#172019621 Admin should be able to add a book category

### PR DESCRIPTION
**Pull request by hakoemmy**
#### What does this PR do?
- Allows an admin to add a book category or section
#### Description of Task to be completed?
-  Create a book section interface using ```React```
- Manage state using ```Redux``` and ```redux thunk```
- Integrate the created front end with a hosted API on this URL: ```https://lms-server-octopus.herokuapp.com/api/v1/sections``` 
#### How should this be manually tested?
- Clone the repo
- Run ```npm run install``` in terminal to get ready all app dependencies
- Run ```npm test``` in the terminal to see if all unit tests are working
- Run ```npm start``` to fire up the local server
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#172019621](https://www.pivotaltracker.com/story/show/172019621)
### Screenshots (if appropriate)
Screen Shot 2020-04-01 at 17 30 12
Screen Shot 2020-04-10 at 04 15 29
Screen Shot 2020-04-10 at 04 15 16

#### Questions:
N/A

### CODE REVIEW

File: src/components/AddBookSection.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import AddBookSectionView from '../views/AddBookSectionView';
+import getUserData from '../utils/jwtHandler';
+
+
+dotenv.config();
+const addBookSection = (props) => {
+    const token = localStorage.getItem('token');
+    const { email, isAdmin } = getUserData(token);
+    return !isAdmin? 
+     (<h1>Here we go with the dashboard for borrowers!</h1>)
```
Comment (blaise82) [inline src/components/AddBookSection.js:14]: i don't get the meaning of this

File: src/hoc/Auxilliary.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const aux = props => props.children;
+
+export default aux;
```
Comment (blaise82) [inline src/hoc/Auxilliary.js]: add a new line

File: src/redux/actionTypes/bookSectionTypes.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const ADD_SECTION_REQUEST = 'ADD_SECTION_REQUEST';
+export const ADD_SECTION_SUCCESS = 'ADD_SECTION_SUCCESS';
+export const ADD_SECTION_FAILURE = 'ADD_SECTION_FAILURE';
```
Comment (blaise82) [inline src/redux/actionTypes/bookSectionTypes.js]: add a new line

File: src/redux/actions/bookSectionAction.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        }).then( res => {
+            const response = res.data.message;
+            dispatch(addSectionSuccess(response));
+        })
+        .catch(error => {
+            const errorMsg = error.response.data.message
+            dispatch(addSectionFailure(errorMsg));
+        })
+    }
+}
```
Comment (blaise82) [inline src/redux/actions/bookSectionAction.js:48]: this is not an action 
I don't know if it should be here

File: src/styles/NavigationItems.module.css [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    flex-flow: column;
+    align-items: center;
+    height: 100%;
+}
+
+@media(min-width: 500px){
+   .NavigationItems{
+    flex-flow: row;
+   }
+}
```
Comment (blaise82) [inline src/styles/NavigationItems.module.css]: new line

File: src/styles/Toolbar.module.css [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    padding: 40px 40px;
+    color: white;
+    font-size: 40px;
+}
+
+@media (max-width: 499px) {
+    .DesktopOnly {
+        display: none;
+    }
+}
```
Comment (blaise82) [inline src/styles/Toolbar.module.css]: new line

File: src/tests/components/home.test.js [hunk 7]
```diff
@@ -1,7 +1,8 @@
+import userEvent from '@testing-library/user-event';
 import React, { Component } from 'react';
 import { render, fireEvent, cleanup, wait, waitForElement } from '@testing-library/react';
 import Home from '../../views/Home';
-import userEvent from '@testing-library/user-event';
```
Comment (blaise82) [inline src/tests/components/home.test.js]: why did you remove it?

File: src/views/Home.js [hunk 8]
```diff
@@ -56,23 +35,49 @@ export default class Home extends Component {
...
+            isLoggedIn: true
+          })
         })
-        .catch((error) => {
+        .catch((err) => {
           this.setState({
-            passwordError: error.response.data.error,
-            emailError: error.response.data.error,
+            passwordError: err.response.data.error,
+            emailError: err.response.data.error,
```
Comment (blaise82) [inline src/views/Home.js]: please stop rewriting my code.

File: src/views/Navigation/NavigationItems/NavigationItem/NavigationItem.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+const navigationItem = (props) =>(
+        <li className={classes.NavigationItem}>
+        <a className={props.active ? classes.active: null} 
+        href={props.link}>{props.children}</a>
+        </li>
+);
+
+
+export default navigationItem;
```
Comment (blaise82) [inline src/views/Navigation/NavigationItems/NavigationItem/NavigationItem.js]: new line

File: src/views/Navigation/NavigationItems/NavigationItems.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    <NavigationItem link="/">ADD BOOK</NavigationItem>
+    <NavigationItem link="/">{props.email}</NavigationItem>
+ </ul>
+);
+
+navigationItems.propTypes = {
+   email: propTypes.string.isRequired
+}
+
+export default navigationItems;
```
Comment (blaise82) [inline src/views/Navigation/NavigationItems/NavigationItems.js]: new line

File: src/views/Navigation/ToolBar/ToolBar.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        <nav className={classes.DesktopOnly}>
+           <NavigationItems email={props.adminEmail}/>
+        </nav>
+    </header>
+);
+ 
+toolbar.propTypes = {
+    adminEmail: propTypes.string.isRequired
+};
+export default toolbar;
```
Comment (blaise82) [inline src/views/Navigation/ToolBar/ToolBar.js]: new line

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: src/components/AddBookSection.js (hunk 1 above)
Comment (hakoemmy) [reply src/components/AddBookSection.js:14]: @blaise82 Thank you for taking the time to comment. Here I am trying to decode the token the user has used to login and if he/she is admin, the system redirect him to the admin dashboard otherwise to normal users dashboard(In our case we don't yet have this page!).

Let me know if you have any suggestions on how we can tweak this?

File: src/hoc/Auxilliary.js (hunk 2 above)
Comment (hakoemmy) [reply src/hoc/Auxilliary.js]: @blaise82 Thank you for FRESH EYES. I solved the thing

File: src/redux/actionTypes/bookSectionTypes.js (hunk 3 above)
Comment (hakoemmy) [reply src/redux/actionTypes/bookSectionTypes.js]: @blaise82 It is resolved

File: src/redux/actions/bookSectionAction.js (hunk 4 above)
Comment (hakoemmy) [reply src/redux/actions/bookSectionAction.js:48]: @blaise82 Thank you again for the suggestion, But the reason why I put this here it 's because ```addSection = (data, token)``` is an async action creator that returns a function instead of an object(This is the power of ```Redux thunk``` middleware).

Let me know if you still have further suggestions on how we can tweak this?

File: src/styles/NavigationItems.module.css (hunk 5 above)
Comment (hakoemmy) [reply src/styles/NavigationItems.module.css]: @blaise82 It's resolved. Thanks

File: src/tests/components/home.test.js (hunk 7 above)
Comment (hakoemmy) [reply src/tests/components/home.test.js]: @blaise82 Thank you. I shifted this on the top due to this ESLint error: ```Absolute imports should come before relative imports```. Are you familiar with this? if you do, you'll better understand what I did.

File: src/views/Home.js (hunk 8 above)
Comment (hakoemmy) [reply src/views/Home.js]: @blaise82 Thank you. The reason why I rewrote your code it's because ESLint was telling me this : ```'error' is already declared in the upper scope``` That means, you had two declarations of the same variable in your ```Home.js``` component.


Let me know if you have some suggestions

File: src/views/Navigation/NavigationItems/NavigationItem/NavigationItem.js (hunk 9 above)
Comment (hakoemmy) [reply src/views/Navigation/NavigationItems/NavigationItem/NavigationItem.js]: @blaise82 It's resolved

File: src/views/Navigation/NavigationItems/NavigationItems.js (hunk 10 above)
Comment (hakoemmy) [reply src/views/Navigation/NavigationItems/NavigationItems.js]: @blaise82 It's resovled

File: src/views/Navigation/ToolBar/ToolBar.js (hunk 11 above)
Comment (hakoemmy) [reply src/views/Navigation/ToolBar/ToolBar.js]: @blaise82 It's well resolved. thanks

**Review by blaise82 — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]
